horovod/common/operations.cc

```diff
@@ -525,6 +533,18 @@ ncclDataType_t GetNCCLDataType(const std::shared_ptr<Tensor> tensor) {
 }
 #endif
 
+#if HAVE_DDL
+DDL_Type GetDDLDataType(const std::shared_ptr<Tensor> tensor) {
+  switch (tensor->dtype()) {
+  case HOROVOD_FLOAT32:
+    return DDL_TYPE_FLOAT;
+  default:
+    throw std::logic_error("Type " + MPIDataType_Name(tensor->dtype()) +
```

Comment:
`std::logic_error()` here is not desirable.

`GetMPIDataType()` and `GetNCCLDataType()` throw `std::logic_error()` because it indicates Horovod developer bug - they both cover all possible types allowed for GPU allreduce & other ops by TF and PyTorch, and if a user tries to use unsupported type he'll get a nice error before that in the stack.

Since here DDL is more restrictive, we should handle it nicer and similarly give a user error via a mechanism similar to `DDL_CHECK()`.

Reply:
I can do that